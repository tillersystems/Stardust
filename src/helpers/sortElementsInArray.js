/**
 * Sort elements in array
 *
 * Compare function in order to sort an array of objects.
 *
 * Function should always return a number, either positive, negative or 0
 * If the return number is negative, A will be shown before B.
 * If the return number is positive, B will be shown before A.
 * If the return number is 0, A and B will remain in the same order as when they entered the loop.
 *
 * @param {string, number} a
 * @param {string, number} b
 *
 * @return {number}
 */

export default function sortElementsInArray(a, b) {
  if (b < a) {
    return -1;
  }
  if (b > a) {
    return 1;
  }
  return 0;
}
