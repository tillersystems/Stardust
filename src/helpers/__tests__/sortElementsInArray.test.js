import sortElementsInArray from '../sortElementsInArray';

describe('sortElementsInArray helpers', () => {
  it('should return 0', () => {
    const expectedValue = sortElementsInArray(1, 1);
    expect(expectedValue).toMatchSnapshot();
  });
  it('should return -1', () => {
    const expectedValue = sortElementsInArray(2, 1);
    expect(expectedValue).toMatchSnapshot();
  });
  it('should return 1', () => {
    const expectedValue = sortElementsInArray(1, 2);
    expect(expectedValue).toMatchSnapshot();
  });
});
