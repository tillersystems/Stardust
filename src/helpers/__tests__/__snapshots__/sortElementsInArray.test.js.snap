// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sortElementsInArray helpers should return -1 1`] = `-1`;

exports[`sortElementsInArray helpers should return 0 1`] = `0`;

exports[`sortElementsInArray helpers should return 1 1`] = `1`;
